superset-frontend/plugins/plugin-chart-handlebars/src/components/Handlebars/HandlebarsViewer.tsx

```diff
@@ -0,0 +1,33 @@
+import { SafeMarkdown } from '@superset-ui/core';
+import Handlebars from 'handlebars';
+import moment from 'moment';
+import React, { useMemo, useState } from 'react';
+
+export interface HandlebarsViewerProps {
+  templateSource: string;
+  data: any;
+}
+
+export const HandlebarsViewer = ({
+  templateSource,
+  data,
+}: HandlebarsViewerProps) => {
+  const [renderedTemplate, setRenderedTemplate] = useState('');
+
+  useMemo(() => {
+    const template = Handlebars.compile(templateSource);
+    const result = template(data);
+    setRenderedTemplate(result);
+  }, [templateSource, data]);
```

Comment:
I would add a try-catch block here to catch compilation errors. Something like
```typescript
const [error, setError] = useState('');

  useMemo(() => {
    try {
      const template = Handlebars.compile(templateSource);
      const result = template(data);
      setRenderedTemplate(result);
      setError('');
    } catch (error) {
      setRenderedTemplate('');
      setError(error.message);
    }
  }, [templateSource, data]);
```

You could then add an Error component that displays the error, like

```typescript
const Error = styled.pre`
  white-space: pre-wrap;
`;

  if (error) {
    return <Error>{error}</Error>;
  }
```

This would produce something like this when the template has a syntax error:

![image](https://user-images.githubusercontent.com/33317356/149485648-7a5fc324-90c2-407d-9e61-519fe858dc84.png)

Reply:
Excellent suggestion!